Sources/NIO/SelectableEventLoop.swift

```diff
@@ -369,11 +371,17 @@ internal final class SelectableEventLoop: EventLoop {
             }
         }
         var nextReadyTask: ScheduledTask? = nil
+        self._tasksLock.withLock {
+            if let firstTask = self._scheduledTasks.peek() {
+                // We're only getting here if something was enqueued before `self.run` was called.
+                nextReadyTask = firstTask
+            }
+        }
```

Comment:
Why do we need this?

Reply:
@Lukasa I'll expand on the comment. But basically: If we enqueue a ready task before `run` is called, we'd block until the next task is enqueued from another thread. This just checks if there's a task already and then only blocks until the ready time for that task.

Technically, this was a race we always had. But given that it's SUPER unlikely that we would publish the `EventLoop` reference before `run` is called, it was impossible to hit :). But now, if you enqueue from the `callback` in `makeEventLoopFromCallingThread`, you can totally do that.